lib/Sema/CSSimplify.cpp

```diff
@@ -6777,9 +6778,11 @@ ConstraintSystem::simplifyCheckedCastConstraint(
   case CheckedCastKind::ArrayDowncast: {
     auto fromBaseType = *isArrayType(fromType);
     auto toBaseType = *isArrayType(toType);
+    auto elementLocator =
+        locator.withPathElement(LocatorPathElt::KnownStdlibCollectionElement());
```

Comment:
Why not just use `GenericArgument(0)`? That's actually what that element type is in type context.

Reply:
Ah sure makes sense I'll adjust it

Reply:
Just adjusted!

Reply:
Great!